model/states.go

```diff
@@ -18,8 +18,29 @@ import (
 	"encoding/json"
 	"fmt"
 	"strings"
+
+	validator "github.com/go-playground/validator/v10"
+	val "github.com/serverlessworkflow/sdk-go/v2/validator"
 )
 
+func init() {
+	val.GetValidator().RegisterStructValidation(stateStructLevelValidation, State{})
+}
+
+func stateStructLevelValidation(structLevel validator.StructLevel) {
+	state := structLevel.Current().Interface().(State)
+
+	hasTransition := state.Transition != nil
+	isEnd := state.End != nil && state.End.Terminate
+
+	// TODO: Improve message errors
+	if !hasTransition && !isEnd {
+		structLevel.ReportError(nil, "State.Transition,State.End", "transition,end", "required", "")
```

Comment:
why using ,State?

Reply:
If using in `BaseState` level, the `SwitchState` try to pass the validation of `BaseState` (is always invalid because it doesn't have the `Transition` or `End`).

In my view all the validations need be in the `State` to check the state `Type` and do the right validation.

What do you think?

Reply:
Changed to `BaseState`.